app/services/partner_reporter_service.rb

```diff
@@ -0,0 +1,17 @@
+module PartnerReporterService
+  def self.partner_types_and_zipcodes(partners:)
+    return [{ 'type1' => 2, 'type2' => 1 }, %w(1111 2222)] if Rails.env.development?
+
+    total = {}
+    zipcodes = []
+
+    partners.each do |partner|
+      partner_agency = DiaperPartnerClient.get(id: partner.id)
+      total[(partner_agency[:agency_type]).to_s] = 0 if total[(partner_agency[:agency_type]).to_s].blank?
+      total[(partner_agency[:agency_type]).to_s] += 1
```

Comment:
Considering you're performing the same action twice, I mean, the action `partner_agency[:agency_type]).to_s`. I believe it worths assigning it to an internal variable.

```ruby
partners.each do |partner|
  partner_agency = DiaperPartnerClient.get(id: partner.id)
  agency_type = partner_agency[:agency_type]).to_s
end
```